src/libraries/Common/src/System/Security/Cryptography/PemKeyHelpers.Factory.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal static partial class PemKeyHelpers
+    {
+        internal delegate TAlg ImportFactoryKeyAction<TAlg>(ReadOnlySpan<byte> source);
+        internal delegate ImportFactoryKeyAction<TAlg>? FindImportFactoryActionFunc<TAlg>(ReadOnlySpan<char> label);
+
+        internal static TAlg ImportFactoryPem<TAlg>(ReadOnlySpan<char> source, FindImportFactoryActionFunc<TAlg> callback) where TAlg : class
```

Comment:
This is largely based on the already existing `ImportPem` from `PemKeyHelpers`. We need a new import method because the new PQC types have a different API shape than the ones on AsymmetricAlgorithm (they are static, and they don't `out` the number of bytes read).

Reply:
I tried de-duping the logic between the two but it got real messy because the delegates are different types and we need type parameters in this one.